### PR TITLE
Remove Identity.auth_type requirement

### PR DESCRIPTION
**Pull request by kruai**
## Overview

This PR is being created to address [RHCLOUD-13924](https://issues.redhat.com/browse/RHCLOUD-13924).
I originally opened this as a hotfix for [RHCLOUD-13572](https://issues.redhat.com/browse/RHCLOUD-13572), as I had mistakenly changed `auth_type` to be required for all Identities instead of only for system-type Identities. However, @lphiri informed us that "auth_type" shouldn't be a required field, so I'm removing that requirement from the Identity class.

## PR Checklist

- [x] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [x] Links to related PRs

## Secure Coding Practices Checklist GitHub Link

- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [x] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

**Review by thearifismail — DISMISSED** [review]

**Review by tahmidefaz — APPROVED** [review]
We really should try and simply this identity logic within HBI. There are so many of them. :pensive:

File: app/auth/identity.py [hunk 1]
```diff
@@ -89,7 +85,9 @@ def __init__(self, obj=None, token=None):
 
             elif obj["type"] == IdentityType.SYSTEM:
                 self.system = obj.get("system")
-                if not self.system:
+                if self.auth_type and self.auth_type not in AuthType.__members__.values():
```
Comment (fstavela) [inline app/auth/identity.py:88]: What about doing this check also for user identity? So, if no `auth_type` is provided, everything is good, but if wrong `auth_type` is provided, raise error regardless of `type`

Comment (kruai) [reply app/auth/identity.py:88]: I'm fine with that. One sec

Comment (thearifismail) [reply app/auth/identity.py:88]: yes, not provided is acceptable and invalid one is BAD

Comment (kruai) [reply app/auth/identity.py]: OK, updated @fstavela

Comment (fstavela) [reply app/auth/identity.py]: looks good!

**Review by fstavela — APPROVED** [review]

**Review by thearifismail — APPROVED** [review]

**Review by tahmidefaz — APPROVED** [review]

**Pull request merged**